File: app-android/core/account/src/main/java/com/aidatainsight/android/core/account/auth/AccountMappers.kt
package com.aidatainsight.android.core.account.auth

import com.aidatainsight.android.core.model.account.AccountSession
import com.aidatainsight.android.core.model.account.AccountUser
import com.aidatainsight.android.core.network.model.OAuthModel

internal fun OAuthModel.toAccountSession(
    username: String? = null,
    previous: AccountSession? = null,
): AccountSession {
    return AccountSession(
        accessToken = accessToken ?: previous?.accessToken,
        refreshToken = refreshToken ?: previous?.refreshToken,
        orgId = orgId ?: previous?.orgId,
        username = username ?: this.username ?: previous?.username,
    )
}

internal fun com.aidatainsight.android.core.model.contract.AccountUser.toAccountUser(): AccountUser {
    return AccountUser(
        id = id,
        phone = phone,
        username = username,
        nickname = nickname,
    )
}

File: app-web/src/data/account/account-mappers.ts
import type { AccountSession, AccountUser } from '@/contracts/generated/models';
import type { OAuthDto } from './account-types';

export function normalizeOAuthSession(
  dto: OAuthDto | null | undefined,
  previous?: AccountSession | null,
): AccountSession {
  const accessToken = normalizeText(dto?.accessToken ?? dto?.access_token) ?? previous?.accessToken ?? null;
  const refreshToken =
    normalizeText(dto?.refreshToken ?? dto?.refresh_token) ?? previous?.refreshToken ?? null;
  const orgId = dto?.orgId ?? dto?.org_id ?? previous?.orgId ?? null;

  return {
    accessToken,
    refreshToken,
    orgId,
    username: previous?.username ?? null,
    isLogin: Boolean(accessToken),
  };
}

export function mergeUserIntoSession(
  session: AccountSession,
  user: AccountUser | null,
): AccountSession {
  return {
    ...session,
    username: user?.username ?? user?.nickname ?? session.username ?? null,
    isLogin: Boolean(session.accessToken),
  };
}

function normalizeText(value: string | null | undefined): string | null {
  const trimmed = value?.trim();
  return trimmed ? trimmed : null;
}
